gwtp-type-slot-upgrader/src/main/java/com/arcbees/gwtp/upgrader/ContentSlotRewriter.java

```diff
@@ -0,0 +1,243 @@
+package com.arcbees.gwtp.upgrader;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import javax.swing.JOptionPane;
+
+import com.github.javaparser.ASTHelper;
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.FieldAccessExpr;
+import com.github.javaparser.ast.expr.MarkerAnnotationExpr;
+import com.github.javaparser.ast.expr.MethodCallExpr;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.ast.expr.ObjectCreationExpr;
+import com.github.javaparser.ast.expr.SuperExpr;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import com.github.javaparser.ast.type.ReferenceType;
+import com.github.javaparser.ast.type.Type;
+import com.github.javaparser.ast.visitor.GenericVisitor;
+import com.github.javaparser.ast.visitor.VoidVisitor;
+
+public class ContentSlotRewriter extends AbstractReWriter {
+    private final static Logger LOGGER = Logger.getGlobal();
+
+    private final static Set<String> slotMethodNames = getSlotMethodNames();
+
+    private final Set<String> allPresenters;
+
+    private final Map<String, Set<String>> slotNames = new HashMap<>();
+
+    // presenters that should be rechecked.
+    private final Set<String> secondRun = new HashSet<>();
+
+    private boolean onSecondRun = false;
+
+    private boolean upgrade;
+
+    public ContentSlotRewriter(Set<String> allPresenters, boolean upgrade) {
+        this.allPresenters = allPresenters;
+        this.upgrade = upgrade;
+    }
+
+    private static Set<String> getSlotMethodNames() {
+        Set<String> smNames = new HashSet<>();
+        smNames.add("addToSlot");
+        smNames.add("clearSlot");
+        smNames.add("removeFromSlot");
+        smNames.add("setInSlot");
+        return smNames;
+    }
+
+    private void processNode(Node node) {
+        if (node instanceof MethodCallExpr) {
+            MethodCallExpr mExpr = (MethodCallExpr) node;
+            if (slotMethodNames.contains(mExpr.getName())) {
+                if (mExpr.getArgs() != null) {
```

Comment:
All the code is unreadable.  I'll try to make it simpler.

Suggest stepping through with a debugger if you're trying to see what happens, though even then this is not nice code.